File: frontend/intellidining-admin-react/src/api/coupon.ts
import { http } from '@/lib/http/request'

export type CouponRow = {
  id: number
  name: string
  discountType: number
  discount: number
  totalCount: number
  remainedCount: number
  beginTime: string
  endTime: string
  createTime?: string
  updateTime?: string
}

export const publishCoupon = (data: {
  name: string
  discountType: number
  discount: number
  totalCount: number
  beginTime: string
  endTime: string
}) =>
  http({
    url: '/coupon/publish',
    method: 'post',
    data,
  })

export const listCoupons = () =>
  http({
    url: '/coupon/list',
    method: 'get',
  })

File: backend/sky-pojo/src/main/java/com/sky/entity/Coupon.java
package com.sky.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Coupon implements Serializable {
    private static final long serialVersionUID = 1L;

    private Long id;
    private String name;
    private Integer discountType;
    private BigDecimal discount;
    private Integer totalCount;
    private Integer remainedCount;
    private LocalDateTime beginTime;
    private LocalDateTime endTime;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

